projects/storefrontapp-e2e-cypress/cypress/support/utils/test-isolation.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+Cypress 12 has introduced Test isolation: https://docs.cypress.io/guides/references/migration-guide#Test-Isolation
+ By default, before each test this mechanism takes place:
+  - Clearing the dom state by visiting about:blank
+  - Clearing cookies in all domains
+  - Clearing localStorage in all domains
+  - Clearing sessionStorage in all domains
+For existing Spartacus e2e tests not supporting TestIsolation, utils functions can simulates pre-Cy12 behavior:
+https://docs.cypress.io/guides/references/migration-guide#Simulating-Pre-Test-Isolation-Behavior
+*/
+
+function clearLocalStorageAndCookies() {
+  cy.clearLocalStorage();
+  cy.clearCookies();
+}
+
+export function isolateTests(): void {
```

Comment:
function containing hooks to simulate pre-cy12 behavior, see guide:
https://docs.cypress.io/guides/references/migration-guide#Simulating-Pre-Test-Isolation-Behavior
in guide, only 'beforeEach' hook is present.
However 'before' also needs to be present to ensure logic in existing e2e test 'before' is not corrupted by cache of previous test.

Many e2e tests have a logic in 'Before' hook (eg: register/login/pageNavigation) without previously clearing cooking/storage.
if previous test (test1) finishes with a diff BasedSite (eg:apparel-uk stored in cookie) or with logged-in user (token in local storage)
The logic inside 'before' hook of next test (test2) will be affected - unwanted basedSite - or unwanted loggedIn user
eg: test1 -> apparel-checkout-flow.e2e : store 'apparel-uk-spa' baseSite get stored in cookie
then
test2-> home-page.core.e2e -> existing 'before' hook navigate to homePage -> Homepage will show as 'apparel-uk-spa' domain